src/main/java/pl/javastart/task/TimeCalculator.java

```diff
@@ -0,0 +1,123 @@
+package pl.javastart.task;
+
+import java.time.LocalDateTime;
+import java.util.*;
+
+public class TimeCalculator {
+
+    public static final String YEAR = "y";
+    public static final String MONTH = "M";
+    public static final String DAY = "d";
+    public static final String HOUR = "h";
+    public static final String MINUTE = "m";
+    public static final String SECOND = "s";
+    public static final String PLUS = "+";
+    public static final String MINUS = "-";
+
+    public static List<DataUnitFromUser> createDateList(String dataInput) {
+        List<DataUnitFromUser> timeUnits = new ArrayList<>();
+        String[] split = dataInput.split("");
```

Comment:
To rozbijanie musisz zrobić trochę sprytniej. Podpowiem, że najlepiej wykorzystać tutaj wyrażenia regularne, ale można też inaczej.
Metoda split pozwala podzielić tekst nie tylko według jakiegoś słowa, czy znaku, ale też według wyrażenia regularnego właśnie. 
Tutaj dobrze jest najpierw podzielić tekst "albo myślnikiem, albo znakiem +", np.:
+20h-4m warto podzielić na:
[20h, 4m]
Później każdy element takiej tablicy można dodatkowo podzielić na liczbę i tekst, np.:
[{20, h}, {4, m}]

Jak już ma się wszystko podzielone, to dalej powinno być już z górki.

Kilka przykładowych fragmentów kodu, które mogą Ci się przydać:
```
String text1 = "+10a-20b+30c";
String[] split1 = text1.split("[+-]");
System.out.println(Arrays.toString(split1));
String replacedText1 = text1.replaceAll("[^+-]", "");
System.out.println(replacedText1);

String text2 = "123abc";
String[] splitted2 = text2.split("(?<=\\d)(?=\\D)");
System.out.println(Arrays.toString(splitted2));
```
Strona, która może być pomocna do testowania różnych rzeczy: https://regex101.com
Fraza, która może być pomocna do wyciągania różnych rzeczy z tekstu "java regex ..."